python/ray/data/dataset.py

```diff
@@ -3935,7 +3936,7 @@ def _is_tensor_dataset(self) -> bool:
             return False
         return schema.names == [TENSOR_COLUMN_NAME]
 
-    def _dataset_format(self) -> str:
+    def dataset_format(self) -> BlockFormat:
         """Determine the format of the dataset. Possible values are: "arrow",
         "pandas", "simple".
```

Comment:
sg, sounds like no objection with exposing `dataset_format` ? This is already the status quo where `_dataset_format` is used and needed outside of its own file in AIR and AIR tests

Reply:
i think this shouldn't be developer API however since people who uses and understands it will mostly be us since we don't expose dataset Block concept to external developers, unlike the batch conversion utils in AIR.

Reply:
>i think this shouldn't be developer API however since people who uses and understands it will mostly be us since we don't expose dataset Block concept to external developers, unlike the batch conversion utils in AIR.

We use the `DeveloperAPI` for APIs used by advanced users **or** other Ray libraries, and since this API is used by the latter, I think that it should have the `DeveloperAPI` annotation.

Besides, without an annotation this inherits the `Dataset` annotation, which is `PublicAPI`.